Lib/asyncio/streams.py

```diff
@@ -329,6 +329,13 @@ def __del__(self):
             closed.exception()
 
 
+def _swallow_unhandled_exception(task):
+    # Do a trick to suppress unhandled exception
+    # if stream.write() was used without await and
+    # stream.drain() was paused and resumed with an exception
+    task.exception()
```

Comment:
Silently ignore exceptions looks like a bad idea. If the task failed, the exception should be at least logged.

Reply:
We have a special case here.
Basically, there are two ways to call `write()` now:
1. Legacy `writer.write(b'data')`
2. Recommended `await writer.write(b'data')`

For the second case, an exception raised by nested `writer.drain()` call is propagated to user.
The first case ignores the exception from nested `writer.drain()` call, allowing existing code to work in a backward compatible way.
A subsequent call to `drain()` raises the exception again, like in the following snippet:
```
writer.write(b'data')  # exception is swallowed
await writer.drain()  # previous exception is raised if any
```

Sorry for such a tricky code.
Ideally, we should make `writer.write()` a coroutine from the very beginning.
Proposed workaround is a strategy to keep backward compatibility and provide a smooth transition path.

Reply:
> writer.write(b'data')  # exception is swallowed

Again, this exception should be logged somehow. It's a bad practice to *silently* ignore exceptions and don't provide any way to the developer to see these exceptions.

Can you try to write the exception?

I only merged your PR only to attempt to fix https://bugs.python.org/issue36870

Reply:
I'm trying to do exactly the opposite than your PR :-) Provide a way to get ignored exceptions: https://github.com/python/cpython/pull/13187 :-)

Reply:
Well, it is not for suppressing an exception entirely, the exception is raised to the caller on `await writer.write()` if this form of call is used.
It is for disabling the exception logging. We do similar things in asyncio somewhere.